scene/2d/box2d_collision_object.cpp

```diff
@@ -379,25 +379,25 @@ Array Box2DCollisionObject::get_colliding_bodies() const {
 }
 
 int Box2DCollisionObject::get_colliding_bodies_fast(Array p_array) const {
+	// TODO if used in the broader scope of this module (beyond just circle/sdf shapes) this may report the same body node multiple times
 	const b2ContactEdge* contact_iterator = body->GetContactList();
 	int i = 0;
-	int size = p_array.size();
-	while (contact_iterator)
-	{	
-		if (contact_iterator->contact->IsTouching())
-		{
-			Box2DCollisionObject* godot_node = contact_iterator->other->GetUserData().owner;
-			if(i < size) {
-				p_array[i] = godot_node;
-				i++;
-			}
-			else {
-				print_error("get_colliding_bodies overload");
-				assert(false);
+	int before_size = p_array.size();
+	while (contact_iterator) {	
+		if (contact_iterator->contact->IsTouching()) {
+			if (i >= p_array.size()) {
+				p_array.resize(p_array.size() * 2);
 			}
+
+			Box2DCollisionObject* godot_node = contact_iterator->other->GetUserData().owner;
+			p_array[i] = godot_node;
+			i++;
 		}
 		contact_iterator = contact_iterator->next;
 	}
+	if (p_array.size() > before_size) {
+		WARN_PRINT("Output buffer size was reallocated from " + itos(before_size) + " to " + itos(p_array.size()));
```

Comment:
Will this print() indicate that it originates from Box2DCollisionObject.get_colliding_bodies_fast ?

Reply:
Yup, file and line number